components/date-picker/generatePicker/interface.ts

```diff
@@ -0,0 +1,93 @@
+import type {
+  PickerRef,
+  PickerProps as RcPickerProps,
+  RangePickerProps as RcRangePickerProps,
+} from 'rc-picker';
+import type { Locale as RcPickerLocale } from 'rc-picker/lib/interface';
+
+import type { InputStatus } from '../../_util/statusUtils';
+import type { AnyObject } from '../../_util/type';
+import type { SizeType } from '../../config-provider/SizeContext';
+import type { Variant } from '../../form/hooks/useVariants';
+import type { TimePickerLocale } from '../../time-picker';
+
+const DataPickerPlacements = ['bottomLeft', 'bottomRight', 'topLeft', 'topRight'] as const;
+
+type DataPickerPlacement = (typeof DataPickerPlacements)[number];
+
+export type PickerLocale = {
+  lang: RcPickerLocale & AdditionalPickerLocaleLangProps;
+  timePickerLocale: TimePickerLocale;
+} & AdditionalPickerLocaleProps;
+
+export type AdditionalPickerLocaleProps = {
+  dateFormat?: string;
+  dateTimeFormat?: string;
+  weekFormat?: string;
+  monthFormat?: string;
+};
+
+export type AdditionalPickerLocaleLangProps = {
+  placeholder: string;
+  yearPlaceholder?: string;
+  quarterPlaceholder?: string;
+  monthPlaceholder?: string;
+  weekPlaceholder?: string;
+  rangeYearPlaceholder?: [string, string];
+  rangeQuarterPlaceholder?: [string, string];
+  rangeMonthPlaceholder?: [string, string];
+  rangeWeekPlaceholder?: [string, string];
+  rangePlaceholder?: [string, string];
+};
+
+type InjectDefaultProps<Props> = Omit<
+  Props,
+  'locale' | 'generateConfig' | 'hideHeader' | 'components'
+> & {
+  locale?: PickerLocale;
+  size?: SizeType;
+  placement?: DataPickerPlacement;
+  /** @deprecated Use `variant` instead */
+  bordered?: boolean;
+  status?: InputStatus;
+  /**
+   * @since 5.13.0
+   * @default "outlined"
+   */
+  variant?: Variant;
+
+  /**
+   * @deprecated `dropdownClassName` is deprecated which will be removed in next major
+   *   version.Please use `popupClassName` instead.
+   */
+  dropdownClassName?: string;
+  popupClassName?: string;
+  rootClassName?: string;
+  popupStyle?: React.CSSProperties;
+};
+
+/** Base Single Picker props */
+export type PickerProps<DateType extends AnyObject = any> = InjectDefaultProps<
+  RcPickerProps<DateType>
+>;
+
+/** Base Range Picker props */
+export type RangePickerProps<DateType extends AnyObject = any> = InjectDefaultProps<
+  RcRangePickerProps<DateType>
+>;
+
+/**
+ * Single Picker has the `multiple` prop,
+ * which will make the `value` be `DateType[]` type.
+ * Here to be a generic which accept the `ValueType` for developer usage.
+ */
+export type PickerPropsWithMultiple<
+  DateType extends AnyObject = any,
```

Comment:
这里默认值用 `AnyObject` 和 `any` 没区别吧？我觉得还是尽量少出现 `any` 比较好
```suggestion
  DateType extends AnyObject = AnyObject,
```

Reply:
默认值其实不重要，只要传了不会用 any

Reply:
我们自己内部消费严格一点。对外的话，兜底定义都用弱类型比较好，这样用户在引用的时候限制比较小。